app/models.py

```diff
@@ -181,9 +182,14 @@ class Meta:
 
 
 class Sub(BaseModel):
-    name = CharField(null=True, unique=True, max_length=32)
+    name = CharField(unique=True, max_length=32)
+    # name can't actually be NULL.
+    # For instance this code (in `deleteannouncement` in app/views/do):
+    #     link=url_for("sub.view_post", sub=post.sid.name, pid=post.pid),
+    # causes an error in the Flask URL construction process when name is None.
```

Comment:
This comment and the others like it in this file belong in the git commit description, not the source code.

Reply:
Thanks, I’ll pull them out.